src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/osgi/OsgiFramework.kt

```diff
@@ -132,6 +134,62 @@ class OsgiFramework(sync: InstanceSync) : InstanceService(sync) {
         enableComponent(pid)
     }
 
+    fun findConfiguration(pid: String): OSGiConfiguration? {
+        return sync.http.get("$CONFIGURATION_PATH/$pid?post=true") { asObjectFromJson(it, OSGiConfiguration::class.java) }
+    }
+
+    fun getConfiguration(pid: String): OSGiConfiguration {
+        aem.logger.debug("Asking for OSGi configuration on $instance")
+        return findConfiguration(pid)
+                ?: throw InstanceException("OSGi Configuration for PID $pid was not found on $instance. Make sure the component exists.")
+    }
+
+    fun getConfigurations(pid: String): List<OSGiConfiguration> {
+        return getConfigList()
+                .filter { pid == it.fpid }
+                .mapNotNull { findConfiguration(it.id) }
+    }
+
+    fun createConfiguration(pid: String, properties: Map<String, Any> = mapOf()) {
+        val currentConfiguration = getConfiguration(pid)
+        val newConfiguration = getConfigPayload(currentConfiguration, properties)
+        sync.http.post("$CONFIGURATION_PATH/$pid", newConfiguration) { response ->
```

Comment:
each http request should be wrapped into `try/catch` so that meaningul error exception will be thrown / showed up to  build user. see other instance services like `PackageManager` / `Repository` to find how to do that... / compose nice message